### PR TITLE
Finish implementation

### PR DESCRIPTION
**Pull request by leoneparise**

### CODE REVIEW

File: walletconnect/src/main/java/com/trustwallet/walletconnect/models/ethereum/WCEthereumSignMessage.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+data class WCEthereumSignMessage (
+    val raw: List<String>,
+    val type: WCMessageType
+) {
+    enum class WCMessageType {
+        MESSAGE, PERSONAL_MESSAGE, TYPED_MESSAGE
+    }
+
+    val data get() = when (type) {
```
Comment (madcake) [inline walletconnect/src/main/java/com/trustwallet/walletconnect/models/ethereum/WCEthereumSignMessage.kt]: Attach protocol description as documentation

File: walletconnect/src/main/java/com/trustwallet/walletconnect/models/WCMethod.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    ETH_PERSONAL_SIGN("personal_sign"),
+    ETH_SIGN_TYPE_DATA("eth_signTypedData"),
+    ETH_SIGN_TRANSACTION("eth_signTransaction"),
+    ETH_SEND_TRANSACTION("eth_sendTransaction"),
+    BNB_SIGN("bnb_sign"),
+    BNB_TRANSACTION_CONFIRM("bnb_tx_confirmation");
+
+    companion object {
+        fun from(method: String): WCMethod? {
+            for (m in values()) {
```
Comment (madcake) [inline walletconnect/src/main/java/com/trustwallet/walletconnect/models/WCMethod.kt]: Why not values().filter()?

File: walletconnect/src/main/java/com/trustwallet/walletconnect/WCInteractor.kt [hunk 3]
```diff
@@ -54,20 +54,18 @@ class WCInteractor (
...
-        val id = helper.id
-        val method = helper.method
-
-        if (method != null) {
-            try {
-                val request = gson.fromJson<JsonRpcRequest<JsonArray>>(payload, typeToken<JsonRpcRequest<JsonArray>>())
-                handleMethod(method, request)
-            } catch (e: Exception) {
-                onFailure(e)
+        val request = gson.fromJson<JsonRpcRequest<JsonArray>>(payload, typeToken<JsonRpcRequest<JsonArray>>())
```
Comment (madcake) [inline walletconnect/src/main/java/com/trustwallet/walletconnect/WCInteractor.kt]: Catch exception

Comment (leoneparise) [reply walletconnect/src/main/java/com/trustwallet/walletconnect/WCInteractor.kt]: Done.

**Pull request merged**